communicator/shared/shared.go

```diff
@@ -0,0 +1,26 @@
+package shared
+
+import (
+	"fmt"
+	"net"
+)
+
+// HostFormatter helps out with host formatting.
+// For instance, the host parameter differs when connecting to an IPv4 address vs. an IPv6 address.
+type HostFormatter interface {
+	Format(string) string
+}
+
+// HostFormatterImpl implements the HostFormatter interface.
+type HostFormatterImpl struct{}
+
+// Format formats the host/IP correctly, so we don't provide IPv6 address in an IPv4 format during node communication.
+func (h *HostFormatterImpl) Format(host string) string {
```

Comment:
No not really. I just usually do it like this since it makes creating tests easier. In this case the tests aren't really taking advantage of this, so in this case it's not needed, unless we want to "overwrite" the Format() function in a test case.

I can change this to be just a single function. What is your opinion on that?